sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1694,4 +1697,355 @@ private[spark] object QueryCompilationErrors {
     new AnalysisException(
       s"Found duplicate column(s) $colType: ${duplicateCol.sorted.mkString(", ")}")
   }
+
+  def noSuchTableError(db: String, table: String): Throwable = {
+    new NoSuchTableException(db = db, table = table)
+  }
+
+  def tempViewNotCachedForAnalyzingColError(tableIdent: TableIdentifier): Throwable = {
+    new AnalysisException(s"Temporary view $tableIdent is not cached for analyzing columns.")
+  }
+
+  def colNotExistError(col: String): Throwable = {
```

Comment:
There exist the function, please refer https://github.com/apache/spark/blob/cd6a4638110ef3f0db8b6366be680870dfb0bcad/sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala#L535